python/ray/data/_internal/arrow_ops/transform_pyarrow.py

```diff
@@ -111,28 +144,11 @@ def concat(blocks: List["pyarrow.Table"]) -> "pyarrow.Table":
                 col = ArrowTensorArray._chunk_tensor_arrays(
                     [chunk for ca in col_chunked_arrays for chunk in ca.chunks]
                 )
```

Comment:
@clarkzinzow just realized, even though we factor out the schema unification logic below in `unify_schemas(...)`, don't we still need to keep this code chunk to calculate `cols` for building the table below?
`table = pyarrow.Table.from_arrays(cols, schema=schema)`

Reply:
Yeah I was referring to a performance bottleneck in terms of having to loop over all the blocks per column while gathering `cols` for building the concatenated table (with `table = pyarrow.Table.from_arrays(cols, schema=schema)`). Do we still need this loop on a column-block level, if we already have taken care of the schema unification logic in the dedicated function?